var/spack/repos/builtin/packages/su2/package.py

```diff
@@ -31,5 +34,55 @@ class Su2(MesonPackage):
     version("7.0.0", sha256="6207dcca15eaebc11ce12b2866c937b4ad9b93274edf6f23d0487948ac3963b8")
     version("6.2.0", sha256="ffc953326e8432a1a6534556a5f6cf086046d3149cfcec6b4e7390eebe30ce2e")
 
+    variant("mpi", default=False, description="enable MPI support")
+    variant("openmp", default=False, description="enable OpenMP support")
+    variant("tecio", default=True, description="enable TECIO support")
+    variant("cgns", default=True, description="enable CGNS support")
+    variant("autodiff", default=False, description="enable AD(reverse) support")
+    variant("directdiff", default=False, description="enable AD(forward) support")
+    variant("pywrapper", default=False, description="enable Python wrapper support")
+    variant("mkl", default=False, description="enable Intel MKL support")
+    variant("openblas", default=False, description="enable OpenBLAS support")
+    variant("mpp", default=False, description="enable Mutation++ support")
+    variant(
+        "mixedprec",
+        default=False,
+        description="use single precision floating point arithmetic for sparse algebra",
+    )
+
+    depends_on("meson@0.61.1:", type=("build"))
     depends_on("python@3:", type=("build", "run"))
     depends_on("zlib")
+    depends_on("pkg-config")
+    depends_on("mpi", when="+mpi")
+    depends_on("swig", when="+pywrapper")
+    depends_on("py-mpi4py", when="+pywrapper")
+    depends_on("intel-oneapi-mkl", when="+mkl")
+    depends_on("openblas", when="+openblas ~mkl")
+    depends_on("cmake", when="+mpp")
+
+    patch("meson_version.patch", when="@7.4.0:7.5.1")
```

Comment:
Thanks, this looks reasonable and since the depends spec for meson is `meson@0.61.1:` that means it should allow any newer version even though 0.61.1 itself is not packaged, right?

I don't have insights on the non-meson related parts of this PR, but your fix to support newer versions of meson LGTM.